File: frontend/src/types/contract.ts
export type ContractStatus = "draft" | "signed" | "cancelled";

export interface Contract {
  id: number;
  public_contract_id: number;
  reference: string;
  date_signature?: string | null;
  garanties?: string | null;
  obligations?: string | null;
  contract_file_url?: string | null;
  statut: ContractStatus;
  created_at: string;
  updated_at: string;
}

export interface ContractCreate {
  public_contract_id: number;
  reference: string;
  garanties?: string;
  obligations?: string;
  contract_file_url?: string;
}

export type ContractUpdate = Partial<ContractCreate> & {
  date_signature?: string;
  statut?: ContractStatus;
};

File: backend/app/models/contract.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Text, func
from sqlalchemy.orm import relationship

from ..database import Base


class Contract(Base):
    __tablename__ = "contracts"

    id = Column(Integer, primary_key=True, index=True)
    public_contract_id = Column(Integer, ForeignKey("public_contracts.id"), nullable=False, unique=True)
    reference = Column(String(120), nullable=False, unique=True)
    date_signature = Column(DateTime(timezone=True), nullable=True)
    garanties = Column(Text, nullable=True)
    obligations = Column(Text, nullable=True)
    contract_file_url = Column(String(512), nullable=True)
    statut = Column(String(50), nullable=False, default="draft")
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)

    public_contract = relationship("PublicContract", back_populates="contract")
